services/skus/service.go

```diff
@@ -1290,7 +1290,6 @@ type credential interface {
 	GetPresentation(context.Context) string
 }
 
-// TODO refactor this see issue #1502
 // verifyCredential - given a credential, verify it.
 func (s *Service) verifyCredential(ctx context.Context, req credential, w http.ResponseWriter) *handlers.AppError {
```

Comment:
yes, it could definitely use further work in terms of refactoring plus there a a few others which have not been mentioned due to the scope, we should probably keep a refactor issue maybe noting the bigger refactoring pieces.

Changes such as the logging seem safe enough to address now as it wont change any logic. Also possibly fixing [this](https://github.com/brave-intl/bat-go/pull/1995#discussion_r1334259094) and some renaming.

The removal of the `http.ResponseWriter` and addressing the error [assertions](https://github.com/brave-intl/bat-go/pull/1995#discussion_r1334276407) are probably best left to a separate issue.